resolver.go

```diff
@@ -151,22 +158,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
```

Comment:
Yes, that is one reading of RFC3484.  But the implementors of all `getaddrinfo` functions I am aware of have applied it even in IPV4-only situations.  

For instance https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/getaddrinfo.c;hb=HEAD#l2443

> getaddrinfo() returns the v4 or v6 addresses in the same order it received from the upstream server

What is the basis for this claim?